src/azure-cli/azure/cli/command_modules/vm/operations/sig_share.py

```diff
@@ -0,0 +1,185 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=no-self-use, line-too-long, protected-access, too-few-public-methods, unused-argument
+from knack.log import get_logger
+
+from azure.cli.core.aaz import register_command, AAZListArg, AAZStrArg
+from azure.cli.core.azclierror import CLIError
+from ..aaz.latest.sig.share import Update as _SigShareUpdate
+from ..aaz.latest.sig import Wait as _SigWait
+
+logger = get_logger(__name__)
+
+
+@register_command(
+    "sig share add",
+)
+class SigShareAdd(_SigShareUpdate):
+    """Share gallery with subscriptions and tenants.
+
+    :example: Share entire gallery with all members of a subscription and/or tenant.
+        az sig share add --resource-group MyResourceGroup --gallery-name MyGallery \\
+        --subscription-ids subId1 subId2 --tenant-ids tenantId1 tenantId2
+    """
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.subscription_ids = AAZListArg(
+            options=["--subscription-ids"],
+            help="A list of subscription ids to share the gallery with.",
+        )
+        subscription_ids = args_schema.subscription_ids
+        subscription_ids.Element = AAZStrArg()
+
+        args_schema.tenant_ids = AAZListArg(
+            options=["--tenant-ids"],
+            help="A list of tenant ids to share the gallery with.",
+        )
+        tenant_ids = args_schema.tenant_ids
+        tenant_ids.Element = AAZStrArg()
+
+        args_schema.operation_type._required = False
+        args_schema.operation_type._registered = False
+        args_schema.groups._required = False
+        args_schema.groups._registered = False
```

Comment:
I'm not sure if we could omit `_required=False` if we set the param to `_registered=False` or not.
```suggestion
        args_schema.operation_type._registered = False
        args_schema.groups._registered = False
```

Reply:
updated, had to keep `operation_type._required=False` because it's a required one in aaz generated code